bindings/java/src/org/sleuthkit/datamodel/OsAccountRealmManager.java

```diff
@@ -63,103 +61,126 @@
 	 * 
 	 * @throws TskCoreException If there is an error creating the realm.
 	 */
-	public OsAccountRealm createRealmByWindowsSid(String sid, Host host) throws TskCoreException {
+	public OsAccountRealm createWindowsRealm(String accountSid, String realmName, Host referringHost, OsAccountRealm.RealmScope realmScope) throws TskCoreException {
 
-		if (Strings.isNullOrEmpty(sid)) {
-			throw new IllegalArgumentException("A SID is required.");
+		if (realmScope == null) {
+			throw new IllegalArgumentException("RealmScope cannot be null. Use UNKNOWN if scope is not known.");
+		}
+		if (referringHost == null) {
+			throw new IllegalArgumentException("A referring host is required to create a realm.");
+		}
+		if (Strings.isNullOrEmpty(accountSid) && Strings.isNullOrEmpty(realmName)) {
+			throw new IllegalArgumentException("Either an address or a name is required to create a realm.");
 		}
 		
-		// get subAuthority sid
-		String subAuthorityId = getSubAuthorityId(sid);
+		Host scopeHost;
+		OsAccountRealm.ScopeConfidence scopeConfidence;
 		
-		// RAMAN TBD: can the SID be parsed in some way to determine local vs domain ??
-			
-		try (CaseDbConnection connection = this.db.getConnection()) {
-			return createRealm("Unknown Domain Name", OsAccountRealm.RealmNameType.INFERRED, subAuthorityId, host, connection);
-		} catch (SQLException ex) {
-			// Create may have failed if the realm already exists. try to get the realm by addr.
-			try (CaseDbConnection connection = this.db.getConnection()) {
-				Optional<OsAccountRealm >accountRealm = this.getRealmByAddr(subAuthorityId, host, connection);
-				if (accountRealm.isPresent()) {
-					return accountRealm.get();
+		switch (realmScope) {
+			case DOMAIN:
+				scopeHost = null;
+				scopeConfidence = OsAccountRealm.ScopeConfidence.KNOWN;
+				break;
+			case LOCAL:
+				scopeHost = referringHost;
+				scopeConfidence = OsAccountRealm.ScopeConfidence.KNOWN;
+				break;
+
+			case UNKNOWN:
+			default:
+				// check if the referring host already has a realm
+				boolean isHostRealmKnown = this.isHostRealmKnown(referringHost);
+				if (isHostRealmKnown) {
+					scopeHost = null;	// the realm does not scope to the referring host since it already has one.
+					scopeConfidence = OsAccountRealm.ScopeConfidence.KNOWN;
 				} else {
-					throw new TskCoreException(String.format("Error creating realm with address = %s", subAuthorityId), ex);
+					scopeHost = referringHost;
+					scopeConfidence = OsAccountRealm.ScopeConfidence.INFERRED;
 				}
-			}
+				break;
+
+		}
+		
+		// RAMAN TBD: can the SID be parsed in some way to determine local vs domain ??
+		
+		// get subAuthority sid
+		String realmAddr = null;
+		String signature;
+		if (!Strings.isNullOrEmpty(accountSid)) {
+			realmAddr = getSubAuthorityId(accountSid);
+			signature = realmAddr;
+		}
+		else {
+			signature = realmName;
 		}
 		
+		// create a realm
+		return createRealm(realmName, realmAddr, signature, scopeHost, scopeConfidence);
```

Comment:
I don't think we should pass in signature. createRealm() should do that by calling updateSignature()